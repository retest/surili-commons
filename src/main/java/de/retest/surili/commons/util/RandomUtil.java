package de.retest.surili.commons.util;

public class RandomUtil {

	public static double requireProbability( final double probability ) {
		if ( probability < 0.0 || probability > 1.0 ) {
			throw new IllegalArgumentException( "The given probability is not in the range [0, 1]: " + probability );
		}
		return probability;
	}

}
