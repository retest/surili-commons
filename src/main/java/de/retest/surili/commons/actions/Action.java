package de.retest.surili.commons.actions;

import java.io.Serializable;
import java.util.Optional;

import de.retest.recheck.ui.descriptors.Element;

public interface Action extends Serializable {

	Optional<Element> getTargetElement();

}
