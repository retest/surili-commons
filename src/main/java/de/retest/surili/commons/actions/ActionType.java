package de.retest.surili.commons.actions;

import java.io.Serializable;
import java.util.Collection;
import java.util.Set;
import java.util.stream.Collectors;

import lombok.NonNull;
import lombok.Value;

/**
 * Every {@link Action} instance belongs to a certain action type. This abstraction helps to limit the number of
 * possible action types in a {@code SutState}. Target elements are represented by the unique path (e.g. XPath) of an
 * element.
 */
@Value
public class ActionType implements Serializable {

	private static final long serialVersionUID = 1L;

	public static final String EMPTY_TARGET_ELEMENT_PATH = "";

	@NonNull
	String elementPath;
	@NonNull
	String type;

	public static ActionType fromAction( final Action action ) {
		final String elementPath = action.getTargetElement()
				.map( element -> element.getIdentifyingAttributes().getPath() ).orElse( EMPTY_TARGET_ELEMENT_PATH );
		final String type = action.getClass().getName();
		return new ActionType( elementPath, type );
	}

	public static Set<ActionType> fromCollection( final Collection<Action> actions ) {
		return actions.stream().map( ActionType::fromAction ).collect( Collectors.toSet() );
	}
}
