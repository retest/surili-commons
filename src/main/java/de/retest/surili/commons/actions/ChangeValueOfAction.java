package de.retest.surili.commons.actions;

import java.util.Optional;

import de.retest.recheck.ui.descriptors.Element;
import lombok.NonNull;
import lombok.Value;

@Value
public class ChangeValueOfAction implements Action {

	private static final long serialVersionUID = 1L;

	@NonNull
	Element targetElement;
	@NonNull
	CharSequence[] keysToSend;

	@Override
	public Optional<Element> getTargetElement() {
		return Optional.of( targetElement );
	}

}
