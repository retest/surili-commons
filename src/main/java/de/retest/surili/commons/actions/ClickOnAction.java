package de.retest.surili.commons.actions;

import java.util.Optional;

import de.retest.recheck.ui.descriptors.Element;
import lombok.NonNull;
import lombok.Value;

@Value
public class ClickOnAction implements Action {

	private static final long serialVersionUID = 1L;

	@NonNull
	Element targetElement;

	@Override
	public Optional<Element> getTargetElement() {
		return Optional.of( targetElement );
	}

}
