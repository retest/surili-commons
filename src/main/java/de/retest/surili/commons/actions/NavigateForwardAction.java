package de.retest.surili.commons.actions;

import java.util.Optional;

import de.retest.recheck.ui.descriptors.Element;
import lombok.Value;

@Value
public class NavigateForwardAction implements Action {

	private static final long serialVersionUID = 1L;

	@Override
	public Optional<Element> getTargetElement() {
		return Optional.empty();
	}

}
