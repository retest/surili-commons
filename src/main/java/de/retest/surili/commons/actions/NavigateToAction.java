package de.retest.surili.commons.actions;

import java.util.Optional;

import de.retest.recheck.ui.descriptors.Element;
import lombok.NonNull;
import lombok.Value;

@Value
public class NavigateToAction implements Action {

	private static final long serialVersionUID = 1L;

	@NonNull
	String url;

	@Override
	public Optional<Element> getTargetElement() {
		return Optional.empty();
	}

}
