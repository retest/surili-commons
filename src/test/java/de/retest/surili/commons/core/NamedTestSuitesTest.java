package de.retest.surili.commons.core;

import java.util.Arrays;
import java.util.LinkedHashSet;
import java.util.Optional;
import java.util.Set;

import org.approvaltests.Approvals;
import org.junit.jupiter.api.Test;

import de.retest.recheck.ui.descriptors.Element;
import de.retest.surili.commons.actions.Action;
import lombok.Value;

class NamedTestSuitesTest {

	@Test
	void should_rename_all_with_unique_names() throws Exception {
		final Action action0 = new MockedAction( 0 );
		final Action action1 = new MockedAction( 1 );

		final TestCase testCase00 = new TestCaseImpl( Arrays.asList( action0, action0 ) );
		final TestCase testCase01 = new TestCaseImpl( Arrays.asList( action1, action1 ) );

		final TestSuite testSuite0 = new TestSuiteImpl( orderedSet( testCase00, testCase01 ) );

		final TestCase testCase10 = new TestCaseImpl( Arrays.asList( action0, action1 ) );
		final TestCase testCase11 = new TestCaseImpl( Arrays.asList( action1, action0 ) );

		final TestSuite testSuite1 = new TestSuiteImpl( orderedSet( testCase10, testCase11 ) );

		final Set<TestSuite> testSuites = orderedSet( testSuite0, testSuite1 );

		final Set<NamedTestSuite> renamed = NamedTestSuites.of( testSuites );

		Approvals.verify( renamed );
	}

	@SuppressWarnings( "unchecked" )
	private <E> Set<E> orderedSet( final E... e ) {
		return new LinkedHashSet<>( Arrays.asList( e ) );
	}

	@Value
	private class MockedAction implements Action {

		private static final long serialVersionUID = 1L;

		private int id;

		@Override
		public Optional<Element> getTargetElement() {
			return Optional.empty();
		}
	}

}
