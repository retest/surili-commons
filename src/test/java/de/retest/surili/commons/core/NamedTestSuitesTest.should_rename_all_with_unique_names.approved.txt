[NamedTestSuiteImpl(testCases=[NamedTestCaseImpl(actions=[de.retest.surili.commons.core.NamedTestSuitesTest$1@0, de.retest.surili.commons.core.NamedTestSuitesTest$1@0], name=TestCase0), NamedTestCaseImpl(actions=[de.retest.surili.commons.core.NamedTestSuitesTest$1@1, de.retest.surili.commons.core.NamedTestSuitesTest$1@1], name=TestCase1)], name=TestSuite0), NamedTestSuiteImpl(testCases=[NamedTestCaseImpl(actions=[de.retest.surili.commons.core.NamedTestSuitesTest$1@0, de.retest.surili.commons.core.NamedTestSuitesTest$1@1], name=TestCase0), NamedTestCaseImpl(actions=[de.retest.surili.commons.core.NamedTestSuitesTest$1@1, de.retest.surili.commons.core.NamedTestSuitesTest$1@0], name=TestCase1)], name=TestSuite1)]
